ext/drx/scatter_gather_aarch64.c

```diff
@@ -48,8 +521,113 @@
 bool
 drx_expand_scatter_gather(void *drcontext, instrlist_t *bb, OUT bool *expanded)
 {
-    /* TODO i#3837: add support for AArch64. */
     if (expanded != NULL)
         *expanded = false;
-    return drmgr_current_bb_phase(drcontext) == DRMGR_PHASE_APP2APP;
+
+    if (drmgr_current_bb_phase(drcontext) != DRMGR_PHASE_APP2APP)
+        return false;
+
+    instr_t *sg_instr = NULL;
+    if (!scatter_gather_split_bb(drcontext, bb, &sg_instr)) {
+        return true;
+    }
+    DR_ASSERT(sg_instr != NULL);
+
+    scatter_gather_info_t sg_info;
+    bool res = false;
+    get_scatter_gather_info(sg_instr, &sg_info);
+
+    if (!(sg_info.is_load && reg_is_z(sg_info.index_reg) &&
+          sg_info.faulting_behavior == DRX_NORMAL_FAULTING))
+        return true;
```

Comment:
I don't understand this: what condition is this that just returns true?  A comment would help.

Reply:
This was checking whether the instruction is one of the ones that is currently supported and returning straight away if not. I have taken it out now because it will be caught further down the function anyway.